notebooks/enterprise-search/app-search-engine-exporter.ipynb

```diff
@@ -0,0 +1,884 @@
+        "\n",
+        "You can find your App Search endpoint and your search private key from the `Credentials` menu inside your App Search instance in Kibana.\n",
+        "\n",
+        "Also note here, we define our `ENGINE_NAME`. For this examplem we are using the `national-parks-demo` sample engine that is available within App Search."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "APP_SEARCH_ENDPOINT = getpass(\"App Search endpoint: \")\n",
+        "APP_SEARCH_PRIVATE_KEY = getpass(\"App Search private key: \")\n",
+        "\n",
+        "app_search = AppSearch(\n",
+        "  APP_SEARCH_ENDPOINT,\n",
+        "  bearer_auth=APP_SEARCH_PRIVATE_KEY\n",
+        ")\n",
+        "\n",
+        "# modify this with your own engine name\n",
+        "ENGINE_NAME = \"national-parks-demo\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "FSsSGl--uqFT"
+      },
+      "source": [
+        "## Export App Search synonyms in Elasticsearch\n",
+        "\n",
+        "To get started with our export, we will first export any [synonyms](https://www.elastic.co/guide/en/app-search/current/synonyms-guide.html) we have in our App Search engine. \n",
+        "\n",
+        "The resulting synonyms will be placed into a new [Elasticsearch synoynm set](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-with-synonyms.html) named the same as our App Search egnine and used in analyzers for our synonyms-filter type later on.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "kpV8K5jHvRK6"
+      },
+      "outputs": [],
+      "source": [
+        "elasticsearch.synonyms.put_synonym(id=ENGINE_NAME, synonyms_set=[])\n",
+        "\n",
+        "for synonym_set in app_search.list_synonym_sets(engine_name=ENGINE_NAME).body['results']:\n",
+        "  elasticsearch.synonyms.put_synonym_rule(\n",
+        "    set_id=ENGINE_NAME,\n",
+        "    rule_id=synonym_set['id'],\n",
+        "    synonyms=\", \".join(synonym_set[\"synonyms\"])\n",
+        "  )"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "lyc-DcjnvTgH"
+      },
+      "source": [
+        "## Export App Search curations in Elasticsearch\n",
+        "\n",
+        "Next, we will export any curations that may be in our App Search engine.\n",
+        "\n",
+        "To export App Search curations we will use Elasticsearch [query rules](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-using-query-rules.html).\n",
+        "At the moment of writing this notebook Elasticsearch query rules only allow for pinning results unlike App Search curations that also allow excluding results.\n",
```

Comment:
There are some size limitations with query rules as well, we may want to think about what happens if they have > 100 or > 1000 curations. 100 is a soft limit that can be configured and 1000 is currently a hard limit (we want to look into this more as part of going out of tech preview)